iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -260,12 +275,14 @@ template <typename T>
 inline T& optional<T>::value() & noexcept
 {
     Expects(has_value());
+    // AXIVION Next Construct AutosarC++19_03-M5.2.8 : The caller ensures this is a valid cast
```

Comment:
The rationale is not so fitting.

The `optional` has the type `T` defined during compile time and the type is unchangeable during the lifetime of the object. All accesses to the underlying data is done via the same static type and therefore the casts are always valid.

Reply:
Sure